tasks.py

```diff
@@ -231,7 +232,7 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None, protocol=None):
 
         with open(manifests_dir + "/metallb.yaml") as f:
             manifest = f.read()
-        manifest = manifest.replace(":main", ":dev-{}".format(architecture))
+        manifest = manifest.replace(":main", ":dev-{}-{}".format(branch, architecture))
```

Comment:
This isn't quite what I had in mind when I filed the bug.  Take this scenario for example:

```
git checkout v0.10
... edit some metallb code ...
inv dev-env
```

At this point, I would expect my dev-env cluster to be running my modified code, but it won't.  It will be running whatever is in quay.io for `main`, because this search/replace won't match anything.  In the `v0.10` branch, the image isn't `:main`.